tests/Faithlife.Analyzers.Tests/NullTestTernariesTests.cs

```diff
@@ -0,0 +1,77 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NUnit.Framework;
+
+namespace Faithlife.Analyzers.Tests
+{
+	[TestFixture]
+	public class NullTestTernariesTests : CodeFixVerifier
+	{
+		[Test]
+		public void ValidUsage()
+		{
+			const string validProgram = c_preamble + @"
+namespace TestApplication
+{
+	public class Person
+	{
+		public string FirstName { get; set; }
+		public string LastName { get; set; }
+	}
+
+	internal static class TestClass
+	{
+		public static void UtilityMethod()
+		{
+			Person person = null;
+			var firstName = person?.FirstName;
+		}
+	}
+}";
+
+			VerifyCSharpDiagnostic(validProgram);
+		}
+
+		[TestCase("var firstName = person.Age != null ? person.FirstName : null;")]
+		[TestCase("var firstName = person.Age == null ? null : person.FirstName;")]
+		[TestCase("var firstName = person.Age.HasValue ? person.FirstName : null;")]
```

Comment:
Should there be a test for the case where the ternary checks `!person.Age.HasValue`?

Reply:
Are these examples of code that would trigger the diagnostic and should be changed? What's the desired replacement? (I don't see how the null-coalescing operator would be used here; what am I missing?)

Reply:
Added better tests here. These were certainly sloppy, tried to get them in quick before I ran out of time at the end of the day, no real excuse.

`var firstName = person.Age?.BirthYear;` <-- This should better catch all the `HasValue` cases. 

I have not removed the `== null` and `!= null` because I still see some value in having them (ease of deployment), but I'm happy to rip those checks out for sure, as they can lead to issues down the road in terms of maintenance or even bugs.

Reply:
FWIW, you can set a PR to Draft status if you know it's not fully-baked.